sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala

```diff
@@ -151,29 +152,34 @@ private[hive] class ClientWrapper(
     // Switch to the initClassLoader.
     Thread.currentThread().setContextClassLoader(initClassLoader)
     val ret = try {
-      val initialConf = new HiveConf(classOf[SessionState])
-      // HiveConf is a Hadoop Configuration, which has a field of classLoader and
-      // the initial value will be the current thread's context class loader
-      // (i.e. initClassLoader at here).
-      // We call initialConf.setClassLoader(initClassLoader) at here to make
-      // this action explicit.
-      initialConf.setClassLoader(initClassLoader)
-      config.foreach { case (k, v) =>
-        if (k.toLowerCase.contains("password")) {
-          logDebug(s"Hive Config: $k=xxx")
-        } else {
-          logDebug(s"Hive Config: $k=$v")
+      val registeredState = SessionState.get
+      if (registeredState != null && registeredState.isInstanceOf[CliSessionState]) {
```

Comment:
I'm not arguing against the change to avoid replacing the SessionState if it already exists.  I'm asking why it has to check to see if it is a `CliSessionState` or not in order for this logic to work.  Ideally we would not couple these components this closely.  If we have to do so, then you need to explain why.